aws-glue/jobs/reporting-ratio_stats.py

```diff
@@ -110,329 +104,246 @@ def run_query_get_result(
 
 # Run run_query_get_result to get file like object ----
 r_file_object = run_query_get_result(
-    athena_client,
-    s3_bucket,
-    SQL_QUERY,
-    athena_db,
-    s3_output,
-    s3_prefix
+    athena_client, s3_bucket, SQL_QUERY, athena_db, s3_output, s3_prefix
 )
 
 # Retrieve s3 location of Athena query result and retrieve it
-target = 's3://'+ s3_bucket + '/' + r_file_object.key
+target = "s3://" + s3_bucket + "/" + r_file_object.key
 
 pull = pd.read_csv(target)
 pull = pull[pull.ratio > 0 & pull.ratio.notnull()]
 
 # Delete all query results for this job from s3 bucket
-response = s3_client.list_objects_v2(Bucket = s3_bucket, Prefix = s3_prefix)
-
-for object in response['Contents']:
-    if re.search("csv", object['Key']):
-        print('Deleting', object['Key'])
-        s3_client.delete_object(Bucket = s3_bucket, Key = object['Key'])
-
+response = s3_client.list_objects_v2(Bucket=s3_bucket, Prefix=s3_prefix)
 
-# COD, PRD, PRB functions ----
-def cod(ratio):
+for object in response["Contents"]:
+    if re.search("csv", object["Key"]):
+        print("Deleting", object["Key"])
+        s3_client.delete_object(Bucket=s3_bucket, Key=object["Key"])
 
-    n = ratio.size
-    median_ratio = ratio.median()
-    cod = 100 / median_ratio * (sum(abs(ratio - median_ratio)) / n)
 
-    return cod
+def median_boot(ratio, nboot=100, alpha=0.05):
+    return boot_ci(np.median, nboot=nboot, alpha=alpha, ratio=ratio)
 
-def prd(fmv, sale_price):
-
-    ratio = fmv / sale_price
-    prd = ratio.mean() / np.average(a = ratio, weights = sale_price)
 
-    return prd
-
-def prb(fmv, sale_price):
-
-    ratio = fmv / sale_price
-    median_ratio = ratio.median()
-
-    lhs = (ratio - median_ratio) /median_ratio
-    rhs = np.log(((fmv / median_ratio) + sale_price) / 2) / np.log(2)
-
-    lhs = np.array(lhs)
-    rhs = np.array(rhs)
-
-    return sm.OLS(lhs, rhs).fit()
-
-# Functions to determine whether assessment fairness criteria has been met
-cod_met = lambda x: 5 <= x <= 15
-
-prd_met = lambda x: 0.98 <= x <= 1.03
+def ccao_median(x):
+    # Remove top and bottom 5% of ratios as per CCAO Data Department SOPs
+    no_outliers = x.between(
+        x.quantile(0.05), x.quantile(0.95), inclusive="neither"
+    )
 
-prb_met = lambda x: -0.05 <= x <= 0.05
+    x_no_outliers = x[no_outliers]
 
-# General boostrapping function
-def boot_ci(fun, *args, nboot = 100, alpha = 0.05):
+    median_n = sum(no_outliers)
 
-    num_args = len(args)
-    args = pd.DataFrame(args).T
-    n = len(args)
+    median_val = np.median(x_no_outliers)
+    median_ci = median_boot(x, nboot=1000)
```

Comment:
@wrridgeway This is copied from the original code, but shouldn't this line be:

```suggestion
    median_ci = median_boot(x_no_outliers, nboot=1000)
```

Reply:
Yes, it should be.

Reply:
@wrridgeway @dfsnow I'm not sure why, but this change seems to make it not run on AWS.